### DISCUSSION TITLE
Lua: interface for setting multiple vim options via table

### DISCUSSION THREAD

**Discussion by LunarLambda** [neovim/neovim#25129] [Ideas]
Just converted my whole init.vim to lua. Happy with it!

The repetition of `vim.o` / `vim.opt` / friends is a little bothersome though.

It would be nice if for options (and maybe similarly, keymaps), we had an interface like this;

```lua
vim.opt.set {
    expandtab = true,
    smarttab = true,
    shiftwidth = 4,
    softtabstop = 8,
    textwidth = 98,
    colorcoloumn = "+2"
}
```

which could be a shorthand for something like

```lua
for k, v in pairs(args) do
    vim.opt[k] = v
end
```

**Comment by clason** [comment]
What's wrong with the loop? This is the intended way; we are not going to add yet more sugar to `vim.opt` (and are in fact trying to get rid of it).

**Comment by LunarLambda** [comment]
well, nothing, but I don't see how adding it would hurt when people are going to write it anyway. also, trying to get rid of it? how so?

**Reply by clason** [reply]
> adding it would hurt

Someone has to write, document, and maintain that code. Features are not free. 

> trying to get rid of it? 

Yes, in favor of `vim.o`. No reason to have two ways of doing the same thing.

**Comment by LunarLambda** [comment]
Does this mean `vim.opt`'s special handling of tables and lists will be removed, or that it will be moved into `vim.o`?

**Reply by clason** [reply]
The plan is to allow options to "natively" handle tables, so special handling will no longer be required (and `vim.o` will support them automatically). At that point (and not before), `vim.opt` will be deprecated.

**Comment by LunarLambda** [comment]
I see, sounds great! Thank you
